mpcstats_lib.py

```diff
@@ -181,18 +181,55 @@ def geometric_mean(data: list[sint]):
 
     return exp2_fx(exponent)
 
+
+def mode(data: list[sint]):
+    n = len(data)
+    data = Array.create_from(data)
+    freqs = sint.Array(n)
+
+    # find frequency of each element in data
+    @for_range(n)
+    def _(i):
+        freqs[i] = sum(if_else(x == data[i], 1, 0) for x in data)
+
+    # set the frequencies of the magic number to zero
+    @for_range(n)
+    def _(i):
+        freqs[i] = sum(if_else(x == MAGIC_NUMBER, 0, freqs[i]) for x in data)
+
+    # find the highest frequency
+    highest_freq = sint(0)
+    @for_range(n)
+    def _(i):
+        highest_freq.update(
+            if_else(freqs[i].greater_than(highest_freq),
+                freqs[i],
+                highest_freq
+        ))
+
+    # get the first elem in data with the highest frequency
+    highest = sint(0)
+
+    # reverse the data to select the first element with the highest
+    # frequency, aligning with the behavior of its python counterpart
+    rev_data = Array.create_from(data.get_reverse_vector())
+
+    @for_range(n)
```

Comment:
I'm thinking we can just range `i` from `n-1` to `0` so we don't need to create another vector `rev_data`. Would this make sense?
```suggestion
    @for_range(n-1, -1, -1)
```

Reply:
Yes. I wasn't aware that such indexing was possible. Fixed the code.